minimesos/src/main/java/com/containersol/minimesos/MesosCluster.java

```diff
@@ -62,6 +62,60 @@ public MesosCluster(ClusterArchitecture clusterArchitecture) {
 
         this.containers = clusterArchitecture.getClusterContainers().getContainers();
         clusterId = Integer.toUnsignedString(new SecureRandom().nextInt());
+        for (AbstractContainer container : containers) {
+            container.setClusterId(clusterId);
+        }
+    }
+
+    /**
+     * Recreate a MesosCluster object based on an existing cluster ID.
+     *
+     * @param clusterId the cluster ID of the cluster that is already running
+     */
+    public MesosCluster(String clusterId) {
+        this.clusterId = clusterId;
+
+        List<Container> containers = dockerClient.listContainersCmd().exec();
+
+        Collections.sort(containers, (c1,c2) -> Long.compare(c1.getCreated(), c2.getCreated()));
+
+        for (Container container : containers) {
+            for (String name : container.getNames()) {
+                if (name.matches("^/minimesos-.+-" + clusterId + "-\\d+$")) {
+                    String[] parts = name.split("-");
+
+                    String role = parts[1];
+                    String uuid = parts[3];
+
+                    switch (role) {
+                        case "zookeeper":
+                            this.containers.add(new ZooKeeper(dockerClient, clusterId, uuid));
+                            break;
+                        case "agent":
+                            this.containers.add(new MesosSlave(dockerClient, clusterId, uuid));
+                            break;
+                        case "master":
+                            this.containers.add(new MesosMaster(dockerClient, clusterId, uuid));
+                            break;
+                        case "marathon":
+                            this.containers.add(new Marathon(dockerClient, clusterId, uuid));
+                    }
+                    break;
+                }
+            }
+        }
+
+        if (containers.isEmpty()) {
+            throw new RuntimeException("No containers found for cluster ID " + clusterId);
+        }
+
+        if (getZkContainer() != null) {
```

Comment:
ZooKeeper is one of the containers in 'containers'. There is no need to run 'docker ps' one more time